### PR TITLE
apply defaults cascade to swatches()

### PR DESCRIPTION
**Pull request by nicolaskruchten**

### CODE REVIEW

**Comment by nicolaskruchten** [comment]
Supersedes #1872 except for the title ... @joelostblom what do you think of this? It now respects `pio.templates.default` and `px.default.templates` and lets you override inline with a `template` kwarg. These swatches basically can't be made as `px.bar` figures so this is probably the next best thing

**Comment by nicolaskruchten** [comment]
And I've fixed the title.

File: packages/python/plotly/_plotly_utils/colors/_swatches.py [hunk 1]
```diff
@@ -1,10 +1,23 @@
-def _swatches(module_names, module_contents):
+def _swatches(module_names, module_contents, template=None):
     """
-    Returns:
+    Parameters
+    ----------
+
```
Comment (joelostblom) [inline packages/python/plotly/_plotly_utils/colors/_swatches.py]:
```suggestion
```

File: packages/python/plotly/_plotly_utils/colors/_swatches.py [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
+    template : str or dict or plotly.graph_objects.layout.Template instance
+        The figure template name or definition.
+
+    Returns
+    -------
+
```
Comment (joelostblom) [inline packages/python/plotly/_plotly_utils/colors/_swatches.py]:
```suggestion
```

File: packages/python/plotly/_plotly_utils/colors/_swatches.py [hunk 3]
```diff
@@ -25,13 +38,13 @@ def _swatches(module_names, module_contents):
...
-            title=module_names,
+            title="plotly.colors." + module_names.split(".")[-1],
             barmode="stack",
             barnorm="fraction",
-            template="plotly",
             bargap=0.5,
             showlegend=False,
             xaxis=dict(range=[-0.02, 1.02], showticklabels=False, showgrid=False),
             height=max(600, 40 * len(sequences)),
+            template=args["template"],
```
Comment (joelostblom) [inline packages/python/plotly/_plotly_utils/colors/_swatches.py]: Could we somehow add control of the width and maybe also height parameter? I had this in the previous PR:

```
height=max(px.defaults.height, 40 * len(sequences)),
width=px.defaults.width,
```

File: packages/python/plotly/_plotly_utils/colors/_swatches.py (hunk 1 above)
Comment (joelostblom) [reply packages/python/plotly/_plotly_utils/colors/_swatches.py]: To make it consistent with how the other px docstrings are written

File: packages/python/plotly/_plotly_utils/colors/_swatches.py (hunk 3 above)
Comment (nicolaskruchten) [reply packages/python/plotly/_plotly_utils/colors/_swatches.py]: We could, but I'm not sure there's such a strong use-case for this... The template thing was quite important to be able to see things against the proper background etc :)

Comment (nicolaskruchten) [reply packages/python/plotly/_plotly_utils/colors/_swatches.py]: Let's do it in another PR?

**Comment by emmanuelle** [comment]
Look good to me, thanks! :dancer:

**Pull request merged**